src/actionsHandler.ts

```diff
@@ -0,0 +1,232 @@
+  tokenLockGlobalSnapshotCond(filter),
+  tokenUnlockGlobalSnapshotCond(filter),
+  allowSpendGlobalSnapshotCond(filter), 
+  spendTxGlobalSnapshotCond(filter)
+]})
+
+export const globalSnapshotActions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { hash_or_ordinal } = event.pathParameters || {};
+    const filter = extractHashOrdinal(hash_or_ordinal);
+
+    return await paginatedQuery(
+      extractPagination(event),
+      toCreatedAtOrdinalCursor,
+      fromCreatedAtOrdinalCursor,
+      { where: { 
+        ...filterByGlobalSnapshot(filter),
+      table_name: { in: dagActionsTables }  
+      }
+      , orderBy: { created_at: 'desc' } },
+      prisma.abstract_transactions_view.findMany,
+      actionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const addressActions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    return await paginatedQuery(
+      extractPagination(event),
+      toCreatedAtOrdinalCursor,
+      fromCreatedAtOrdinalCursor,
+      { where: { source_addr: address, table_name: { in: dagActionsTables } }, 
+      orderBy: { created_at: 'desc' } },
+      prisma.abstract_transactions_view.findMany,
+      actionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+
+
+const metagraphIdCond = (metagraph_id) => ({ OR: [
+    { metagraph_token_lock: {metagraph_id }},
+    { metagraph_token_unlock: {metagraph_id }},
+    { metagraph_allow_spend: {metagraph_id }},
+    { metagraph_spend_transaction: {metagraph_id }},
+    { metagraph_fee_transaction: {metagraph_id }}
+  ]})
+
+const tokenLockMetagraphSnapshotCond = (filter) => ({ metagraph_token_lock:  { metagraph_token_lock_block: { metagraph_snapshot: filter} } })
+const tokenUnlockMetagraphSnapshotCond = (filter) => ({ metagraph_token_unlock:  tokenLockMetagraphSnapshotCond(filter) })
+const allowSpendMetagraphSnapshotCond = (filter) => ({
+  metagraph_allow_spend: { 
+    metagraph_allow_spend_block: {
+      metagraph_snapshot: filter
+    }
+  }
+})
+const spendTxMetagraphSnapshotCond = (filter) => ({ metagraph_spend_transaction: allowSpendMetagraphSnapshotCond(filter) })
+
+
+const filterByMetagraphSnapshot = (filter) => ( {OR: [
+  tokenLockMetagraphSnapshotCond(filter),
+  tokenUnlockMetagraphSnapshotCond(filter),
+  allowSpendMetagraphSnapshotCond(filter), 
+  spendTxMetagraphSnapshotCond(filter)
+]})
+
+
+export const currencyActions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { metagraph_id } = event.pathParameters || {};
+
+    return await paginatedQuery(
+      extractPagination(event),
+      toCreatedAtOrdinalCursor,
+      fromCreatedAtOrdinalCursor,
+      { where: { 
+        ...metagraphIdCond(metagraph_id),
+        table_name: { in: metagraphActionsTables } 
```

Comment:
It queries 5 tables at once? How does that work internally? Does it make 5 separate queries or does it somehow use the abstract tables? 

It seems worrying for performance reasons.

Reply:
The view ends up just being an alias for this query so we're joining all these tables together on every query of the view. It should be good enough for now but we should think about optimizations in the future.

Reply:
maybe with two tables  (tx_hash, metagraph_id, metagraph_snapshot_hash), and (tx_hash,  global_snapshot_hash)   we can avoid all the joins